code/datums/components/lockable_storage.dm

```diff
@@ -141,17 +143,45 @@
 	source.balloon_alert(user, "hacked")
 	lock_code = null
 
+/datum/component/lockable_storage/proc/on_emag(obj/source, mob/user, obj/item/card/emag/emag_card)
+	SIGNAL_HANDLER
+
+	if(!source.atom_storage.locked)
+		return FALSE
+
+	if(source.obj_flags & EMAGGED)
+		return FALSE
+	source.obj_flags |= EMAGGED
+	can_hack_open = FALSE // since it's broken for good
+
+	source.visible_message(span_warning("Sparks fly from [source]!"), blind_message = span_hear("You hear a faint electrical spark."))
+	source.balloon_alert(user, "lock destroyed")
+	playsound(source, SFX_SPARKS, 50, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+	lock_code = null
+	source.atom_storage.locked = STORAGE_NOT_LOCKED
+	source.update_appearance()
+	return ITEM_INTERACT_SUCCESS
+
 ///Updates the icon state depending on if we're locked or not.
 /datum/component/lockable_storage/proc/on_update_icon_state(obj/source)
 	SIGNAL_HANDLER
-	source.icon_state = "[source.base_icon_state][source.atom_storage.locked ? "_locked" : null]"
+
+	if(source.obj_flags & EMAGGED)
+		source.icon_state = "[source.base_icon_state]_broken"
+	else
+		source.icon_state = "[source.base_icon_state][source.atom_storage.locked ? "_locked" : null]"
 
 ///Called when interacted with in-hand or on attack, opens the UI.
 /datum/component/lockable_storage/proc/on_interact(atom/source, mob/user)
 	SIGNAL_HANDLER
 	INVOKE_ASYNC(src, PROC_REF(ui_interact), user)
 
 /datum/component/lockable_storage/ui_interact(mob/user, datum/tgui/ui)
+	var/obj/source = parent
+	if(source.obj_flags & EMAGGED)
+		ui.close()
```

Comment:
```suggestion
		ui?.close()
```

Reply:
Also can you swap out the label with a fix and remove feature since I dug through git blame and discovered this already did exist.